pkg/manifest/context.go

```diff
@@ -0,0 +1,73 @@
+package manifest
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/kartverket/skipctl/pkg/git"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+// Context holds shared resources for manifest operations.
+// It lazily loads the git filesystem only when needed.
+type Context struct {
+	ref       string
+	gitDir    string
+	currentFS filesys.FileSystem
+	mu        sync.Mutex
+	gitDirErr error
+}
+
+// NewContext creates a new context for manifest operations.
+func NewContext(ref string) *Context {
+	return &Context{
+		ref:       ref,
+		currentFS: filesys.MakeFsOnDisk(),
```

Comment:
Har du tenkt noe på bruken av `filesys.MakeFsOnDisk()` og avhengigheten til kustomize bibliotek på for en "generisk" context?

Er det nødvendig eller bare praktisk?